File: backend/app/models/schemas/assets.py
from __future__ import annotations
from pydantic import BaseModel, Field, HttpUrl
from typing import Optional, List, Dict, Any
from datetime import datetime

class BBox(BaseModel):
    # For pdffigures2 later; keep nullable for MVP
    page: Optional[int] = None
    x1: Optional[float] = None
    y1: Optional[float] = None
    x2: Optional[float] = None
    y2: Optional[float] = None

class Asset(BaseModel):
    id: str
    kind: str = "image"
    upload_id: str
    filename: str
    rel_path: str                  # path relative to repo root (or to a configured base)
    width: Optional[int] = None
    height: Optional[int] = None
    ext: Optional[str] = None
    checksum: Optional[str] = None
    bbox: Optional[BBox] = None
    caption: Optional[str] = None
    tags: List[str] = Field(default_factory=list)
    created_at: datetime = Field(default_factory=datetime.utcnow)

class AssetList(BaseModel):
    items: List[Asset]
    count: int

File: frontend/src/lib/assets.ts
import { API_BASE } from "./api";

export type Asset = {
  id: string;
  filename: string;
  rel_path: string;
  width?: number;
  height?: number;
  ext?: string;
  checksum?: string;
  caption?: string | null;
};

export type AssetListResp = { items: Asset[]; count: number };

export async function listAssets(uploadId: string): Promise<Asset[]> {
  const base = String(API_BASE).replace(/\/$/, "");
  const res = await fetch(`${base}/assets?upload_id=${encodeURIComponent(uploadId)}`);
  if (!res.ok) throw new Error(`List assets failed: ${res.status}`);
  const data = (await res.json()) as AssetListResp;
  return data.items ?? [];
}

export function assetFileUrl(uploadId: string, assetId: string): string {
  const base = String(API_BASE).replace(/\/$/, "");
  return `${base}/assets/${assetId}/file?upload_id=${encodeURIComponent(uploadId)}`;
}
